pandas/_libs/lib.pyx

```diff
@@ -800,8 +801,30 @@ cpdef ndarray[object] ensure_string_array(
         # short-circuit, all elements are str
         return result
 
+    if arr.dtype.kind == "f":  # non-optimized path
```

Comment:
This comment wasn't clear to me - I think you're saying "Could be optimized more". Would that be better?

Reply:
Similar to the fix for non-float cases; arr PyObject at this point, so every loop iteration it could have to spend time checking that the object is any array, instead of validating that outside the loop and trusting. 

The object case was optimized through the assignment to an ndarray[object] - this would be faster if we did something similar with floats I just didn't think it was worth using fused types for this edge case

Reply:
Sure - understood. I'm just saying I find the comment is unclear (non-optimized how?). Though I see now my suggestion isn't any better. But I'm fine with this as-is.